python_modules/libraries/dagster-dg/dagster_dg/cli/scaffold.py

```diff
@@ -121,6 +124,30 @@ def _define_commands(self, cli_context: click.Context) -> None:
 
         self._commands_defined = True
 
+    def _get_matching_command(self, ctx: click.Context, input_cmd: str) -> click.Command:
+        commands = self.list_commands(ctx)
+        cmd_query = input_cmd.lower()
+        matches = sorted([name for name in commands if cmd_query in name.lower()])
+
+        if not matches:
+            exit_with_error(generate_missing_plugin_object_error_message(input_cmd))
+
+        # Present a menu of options for the user to choose from
+        click.echo(f"No exact match found for '{input_cmd}'. Did you mean one of these?")
+        for i, match in enumerate(matches, 1):
+            click.echo(f"{i}. {match}")
+
+        # Get user selection
+        selection = click.prompt("Select an option (number)", type=int, default=1)
+        if selection < 1 or selection > len(matches):
+            exit_with_error(
+                f"Invalid selection: {selection}. Please choose a number between 1 and {len(matches)}."
+            )
```

Comment:
I think we should have a quit option `n`?

Also in the case where there is only one match I think a `[y/n]` selection would be nicer.